pokete_general_use_fns.py

```diff
@@ -21,5 +21,10 @@ def hard_liner(l_len, name):
     return ret
 
 
+def sort_vers(vers):
+    return [k[-1] for k in
+            sorted([([int(j) for j in i.split(".")], i) for i in vers])]
```

Comment:
Do you plan on using other parts of semantic versioning, such as `beta`, `alpha` and so on? If yes, this algorithm wouldn't work. I tried to provide a doable way below, but it can be compressed with some list comprehensions at some point, if you want to use it.

```python
# New function also sorts version according to their suffix.
# Versions with no suffix will be treated as older, than versions with a suffix.
# Also beta is newer than alpha and so on...
def sort_vers(vers):
    out = []
    for i in vers:
        dec_split = i.split(".")
        strings = []
        for ii in dec_split:
            dash_split = ii.split("-")
            if len(dash_split) == 1:
                strings.append(int(dash_split[0]))
            else:
                for iii in dash_split:
                    try:
                        strings.append(int(iii[0]))
                    except ValueError:
                        # If there is text in this field: try to parse it and give it a number. These suffixes are older then a new '.-release'
                        if "dev" == iii.lower() or "development" == iii.lower() or "SNAPSHOT" == iii.upper():
                            strings.append(-4)
                        elif "alpha" == iii.lower():
                            strings.append(-3)
                        elif "beta" == iii.lower():
                            strings.append(-2)
                        elif iii.lower().startswith("rc"):
                            strings.append(-1)
                        else:
                            strings.append(-5)
        out.append((i, strings))
    sorted_out = sorted(out, key=lambda x: x[1])

    # Strips the temporary placeholders at index 1 away.
    out = []
    for i in sorted_out:
        out.append(i[0])
    return out
```

Reply:
Also: Here is a sample-output of this function:

```python
print(sort_vers(["1.2.0", "1.2.1", "1.1.0", "1.2.0-beta", "1.2.0-alpha", "1.2.0-alpha-1", "1.2.0-alpha-2", "1.2.0-alpha-2.0", "1.2.0-alpha-2.1"]))

['1.1.0', '1.2.0', '1.2.0-alpha', '1.2.0-alpha-1', '1.2.0-alpha-2', '1.2.0-alpha-2.0', '1.2.0-alpha-2.1', '1.2.0-beta', '1.2.1']
```

Reply:
I don't plan on using non numeric versions, that's why I just implemented a numeric sorting.
But thanks for your suggestion.